src/channels/cave-story/index.tsx

```diff
@@ -0,0 +1,389 @@
+import type { FormattedDonation, Total } from '@gdq/types/tracker';
+import { ChannelProps, registerChannel } from '../channels';
+
+import { useListenFor, useReplicant } from 'use-nodecg';
+import styled from '@emotion/styled';
+import { css, keyframes } from '@emotion/react';
+import TweenNumber from '@gdq/lib/components/TweenNumber';
+
+import { useEffect, useRef } from 'react';
+import React from 'react';
+import { useRafLoop } from 'react-use';
+
+import clouds1 from './images/clouds1.png';
+import clouds2 from './images/clouds2.png';
+import clouds3 from './images/clouds3.png';
+import clouds4 from './images/clouds4.png';
+import clouds5 from './images/clouds5.png';
+import dragon from './images/dragon.gif';
+import sue from './images/sue.gif';
+import level from './images/level.png';
+import balrog from './images/balrog.gif';
+import ghostcat from './images/ghostcat.gif';
+import bat from './images/bat.gif';
+import critterSmall from './images/critterSmall.gif';
+import critterBig from './images/critterBig.gif';
+import misery from './images/misery.gif';
+import puppy from './images/puppy.gif';
+import quote from './images/quote.gif';
+import quoteMask from './images/quoteMask.gif';
+import curly from './images/curly.gif';
+
+registerChannel('Cave Story', 290, CaveStory, {
+	position: 'bottomLeft',
+	site: 'GitHub',
+	handle: 'omgitsraven',
+});
+
+function CaveStory(props: ChannelProps) {
+	const [total] = useReplicant<Total | null>('total', null);
+
+	const flyingDonations = useRef<Set<DonationFlyerData>>(new Set<DonationFlyerData>());
```

Comment:
This will create and immediately discard an empty set every time this component is rendered after the initial render, which isn't the worst, but may be something to consider avoiding.

Currently, there is a bug that the spawned flyers do not get removed from the DOM until the next donation *after* they've reached the end of their animation. This is because updating a ref does not cause a re-render, so removing a flyer from the set only takes effect on the next re-render (which for this component, is when the total updates).

This only is visible to the viewer in the specific case of the images that have bullets shooting downwards, where the bullets linger in the bottom left corner. Theoretically this can just be fixed by having the elements final position be fully off-screen, and I'd accept that solution. A more "proper" solution would be to rely on a useState and to update an array inside, but would be more involved to reconfigure everything.